src/Microsoft.AspNet.Identity/UserManager.cs

```diff
@@ -1063,10 +1062,9 @@ public virtual Task<IdentityResult> RemoveClaimAsync(TUser user, Claim claim)
                 throw new ArgumentNullException("roles");
             }
 
-            var userRoles = await userRoleStore.GetRolesAsync(user, CancellationToken);
             foreach (var role in roles)
```

Comment:
Well UserRoles is a db set and I'm guessing Any is checking against the modified (but not saved) set of roles which is why it catches it.  So we are just relying on the store to do the right thing in the manager.  We could explicitly filter with a Distinct if you want.  Not a big deal, just a question if we want an exception or a silent distinct.